gcc/rust/checks/errors/borrowck/ffi-polonius/src/lib.rs

```diff
@@ -172,9 +192,54 @@ pub unsafe extern "C" fn polonius_run(
         }
     }
 
+    let loan_errors = FFIVector__new_vec_pair(output.errors.len());
+    for (keys, values) in output.errors.iter() {
+        let loans_vec = FFIVector__new(values.len());
+        for loan in values.iter() {
+            let loan: usize = (*loan).into();
+            FFIVector__push(loans_vec, loan);
+        }
```

Comment:
I'm not a fan of that `(*loan).into()` syntax we have to use. I would like you to add this extra implementation at the top of the file, alongside the impl for `From<GccrsAtom> for usize`:

```rust
impl From<GccrsAtom> for usize {
    fn from(atom: GccrsAtom) -> Self {
        atom.index()
    }
}

impl From<&GccrsAtom> for usize {
    fn from(atom: &GccrsAtom) -> Self {
        atom.index()
    }
}
```

which will then enable us to do this instead:

```suggestion
        for loan in values.iter() {
            FFIVector__push(loans_vec, loan.into());
        }
```

Reply:
this will also help in removing all of the other `(*<something>).into()`